content/en/docs/concepts/workloads/pods/pod-lifecycle.md

```diff
@@ -460,21 +460,180 @@ An example flow:
+   should not continue to serve regular traffic and should be terminating connection.
+   Some pods require the graceful termination - for example session draining and completion. Without the feature
+   gate `EndpointSliceTerminatingCondition` (in beta starting Kubernetes version `1.22`),
+   load balancers (like the service proxy) remove the Pod from
+   the list of Endpoints in EndpointSlices as soon as the termination grace period _begins_. With the feature
+   gate `EndpointSliceTerminatingCondition`, Endpoints are not removed from EndpointSlices for the terminating pods
+   and a status indicating [terminating state](/docs/concepts/services-networking/endpoint-slices/#conditions)
+   is exposed from Endpoint API. Terminating Endpoints always have `ready` status
+   as `false` for backward compatibility, so load balancers will not use it for regular traffic.
+   If traffic draining on terminating pod is needed, the actual readiness can be checked as a condition `serving`.
 1. When the grace period expires, the kubelet triggers forcible shutdown. The container runtime sends
    `SIGKILL` to any processes still running in any container in the Pod.
    The kubelet also cleans up a hidden `pause` container if that container runtime uses one.
 1. The kubelet triggers forcible removal of Pod object from the API server, by setting grace period
    to 0 (immediate deletion).
 1. The API server deletes the Pod's API object, which is then no longer visible from any client.
 
+
+#### Example flow with endpoint termination
+
+The following is the example of the flow described in previous section.
+
+Let's say you have a Deployment containing of a single `nginx` replica
+(just for demonstration purposes) and a Service:
+
+``` yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment
+  labels:
+    app: nginx
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: nginx
+  template:
+    metadata:
+      labels:
+        app: nginx
+    spec:
+      terminationGracePeriodSeconds: 120
+      containers:
+      - name: nginx
+        image: nginx:latest
+        ports:
+        - containerPort: 80
+        lifecycle:
+          preStop:
+            exec:
+              # just hang around for the duration of terminationGracePeriodSeconds
+              command: [
+                "/bin/sh", "-c", "sleep 120"
+              ]
+
+---
+
+apiVersion: v1
+kind: Service
+metadata:
+  name: nginx-service
+spec:
+  selector:
+    app: nginx
+  ports:
+    - protocol: TCP
+      port: 80
+      targetPort: 80
+```
+
+Once the Pod and Service are running, you can get the name of EndpointSlice:
+
+``` sh
+kubectl get endpointslice
+
+NAME                  ADDRESSTYPE   PORTS   ENDPOINTS                 AGE
+nginx-service-6tjbr   IPv4          80      10.12.1.199,10.12.1.201   22m
+```
+
+You can see it's status and validate that there is one Endpoint registered:
+
+``` sh
+> kubectl get endpointslice -o json nginx-service-6tjbr
```

Comment:
Please [separate commands from output](https://kubernetes.io/docs/contribute/style/style-guide/#separate-commands-from-output) (and also, don't include a prompt)